doc/src/Howto_lammps_gui.rst

```diff
@@ -533,13 +535,32 @@ keyboard, the first of those entries is chosen.
 
 If the word under the cursor is a file, then additionally the context
 menu has an entry to open the file in a read-only text viewer window.
-This is a convenient way to view the contents of files that are
-referenced in the input.  The file viewer also supports on-the-fly
+If the file is a LAMMPS restart file, instead the menu entry offers to
+:ref:`inspect the restart <inspect_restart>`.
+
+The text viewer is a convenient way to view the contents of files that
+are referenced in the input.  The file viewer also supports on-the-fly
 decompression based on the file name suffix in a :ref:`similar fashion
 as available with LAMMPS <gzip>`.  If the necessary decompression
 program is missing or the file cannot be decompressed, the viewer window
 will contain a corresponding message.
 
+.. _inspect_restart:
+
+Inspecting a Restart file
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. versionadded:: 1.6
+
+When LAMMPS-GUI is asked to "Inspect a Restart", it will read the
+restart file into a LAMMPS instance and then open three different
+windows.  The first window is a text viewer with the output of an
+:doc:`info command <info>` with system information stored in the
+restart.  The second window is text viewer containing a data file
+generated with a :doc:`write_data command <write_data>`.  The third
+window is a :ref:`Snapshot Image Viewer <snapshot_viewer>` containing a
+visualization of the system in the restart.
```

Comment:
I'm not familiar with how the image viewer works, but I'm just curious, how demanding could this be if you unintentionally inspected a restart file from a very large system on, say, a laptop?

Reply:
Good point. That could be a problem. I would expect less of a problem from the image, but from converting the restart to a data file and then show that. The entire file is stored in memory. 

It is easy to add a check for the file size and then define some threshold and have a dialog "You are trying to inspect a very large restart, do you really, *really*, **REALLY** want that?" or something like that.

I've never run a LAMMPS simulation big enough where that made a difference, but I've also always been working on machines with plenty of RAM.

Does anybody have an idea of a meaningful cutoff value? 1GB? 25% RAM? 5% RAM?

Reply:
What if you opened the first window, let the user know how many atoms there were, then presented a button which could be clicked to open windows 2 & 3?

Although, it might get old if you have to constantly click an extra confirmation button for small systems...

Reply:
By that point you would have already read the entire restart into memory. And the number of atoms is not that reliable a measure, since the storage required can change a lot depending on the atom style: "atomic" has far fewer fields that need to be allocated and filled than, say "full".

Reply:
Ah, if you have to read the whole restart file to get natoms that defeats the purpose of my suggestion. Could have been useful to quickly access high level details of a large restart file even on an underpowered machine.

I'd vote for 1 GB as a fair warning - if it's too big for notepad maybe a data file shouldn't be opened in text.

Reply:
I'll make some experiments with the rhodo benchmark. Have it write out a restart file and compare memory used by LAMMPS after reading it with a single processor, size of data file and then see what size of restart file will require a large chunk of RAM. I think 1GB on the restart file might be too large. I think 1GB RAM could be ok assuming that even underpowered devices have around 4GB RAM these days (hell, even smartphones have that).

Reply:
As expected with atom style atomic the memory consumption is less. So for a restart file of about 250MByte containing 3M atoms, the memory consumption for reading the restart and viewing the data file is a bit under 1GByte RAM. For atom style full with the rhodo benchmark, I need about 1.2GB RAM for 1M atoms and the same size restart file. So the warning limit is now empirically determined to be at restart files with over 250MBytes.

Reply:
done